parser/src/parser.cpp

```diff
@@ -225,24 +250,64 @@ int main(int argc, char* argv[])
   if (vm.count("force"))
     cc::util::removeTables(db, SQL_DIR);
 
-  cc::util::createTables(db, SQL_DIR);
+  if (vm.count("force") || isNewDb)
+    cc::util::createTables(db, SQL_DIR);
 
   //--- Start parsers ---//
 
   cc::parser::SourceManager srcMgr(db);
   cc::parser::ParserContext ctx(db, srcMgr, compassRoot, vm);
   pHandler.createPlugins(ctx);
 
+  // TODO: Handle errors returned by preparse().
+  std::vector<std::string> topologicalOrder = pHandler.getTopologicalOrder();
+  for (auto it = topologicalOrder.rbegin(); it != topologicalOrder.rend(); ++it)
+  {
+    LOG(info) << "[" << *it << "] preparse started!";
+    if(!pHandler.getParser(*it)->preparse())
```

Comment:
Space missing between if and paren.

Reply:
Fixed in 64cb407581662e78675baf916a95dba2b765b3e4.